src/Compilers/CSharp/Test/CommandLine/SarifV2ErrorLoggerTests.cs

```diff
@@ -364,5 +370,203 @@ public void AnalyzerDiagnosticsSuppressedWithNullJustification()
 
             return string.Format(CultureInfo.InvariantCulture, s, arguments);
         }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void AnalyzerDisabledWithCommandLineOptions()
+        {
+            var source = @"
+class C
+{
+}";
+            var sourceFile = Temp.CreateFile().WriteAllText(source).Path;
+            var errorLogDir = Temp.CreateDirectory();
+            var errorLogFile = Path.Combine(errorLogDir.Path, "ErrorLog.txt");
+
+            string[] arguments = new[] { "/nologo", "/t:library", "/nowarn:ID1", "/nowarn:ID2", sourceFile, "/preferreduilang:en", $"/errorlog:{errorLogFile}{ErrorLogQualifier}" };
+
+            var cmd = CreateCSharpCompiler(null, WorkingDirectory, arguments,
+               analyzers: new[] { new AnalyzerForErrorLogTest() });
+            var outWriter = new StringWriter(CultureInfo.InvariantCulture);
+
+            var exitCode = cmd.Run(outWriter);
+            var actualConsoleOutput = outWriter.ToString().Trim();
+
+            // Assert no diagnostics are reported as the analyzer has been disabled with nowarn.
+            Assert.DoesNotContain("Category1", actualConsoleOutput);
+            Assert.DoesNotContain("Category2", actualConsoleOutput);
+            Assert.Equal(0, exitCode);
+
+            var actualOutput = File.ReadAllText(errorLogFile).Trim();
+
+            var expectedOutputMarkup =
+@"{{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.1.0"",
+  ""version"": ""2.1.0"",
+  ""runs"": [
+    {{
+      ""results"": [
+      ],
+      ""tool"": {{
+        ""driver"": {{
+          ""name"": ""{0}"",
+          ""version"": ""{1}"",
+          ""dottedQuadFileVersion"": ""{2}"",
+          ""semanticVersion"": ""{3}"",
+          ""language"": ""{4}"",
+{5}
+        }}
+      }},
+      ""columnKind"": ""utf16CodeUnits""
+    }}
+  ]
+}}";
+            var expectedOutput = FormatOutputText(
+                expectedOutputMarkup,
+                cmd,
+                AnalyzerForErrorLogTest.GetExpectedV2ErrorLogRulesText(
+                    suppressionKinds1: new[] { "external" }, suppressionKinds2: new[] { "external" }));
+
+            Assert.Equal(expectedOutput, actualOutput);
+
+            CleanupAllGeneratedFiles(sourceFile);
+            CleanupAllGeneratedFiles(errorLogFile);
+        }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void AnalyzerPartiallyDisabledWithEditorconfig()
+        {
+            var source1 = @"
+[System.Diagnostics.CodeAnalysis.SuppressMessage(""Category2"", ""ID1"", Justification = ""Justification1"")]
+[System.Diagnostics.CodeAnalysis.SuppressMessage(""Category2"", ""ID2"", Justification = ""Justification2"")]
+class C
+{
+}";
+            var source2 = @"
+class C2
+{
+}";
+            var editorconfigText = @"
+[source2.cs]
+dotnet_diagnostic.ID1.severity = none
```

Comment:
>ID1 [](http://example.com/codeflow?start=18&length=3)

It looks like there is one suppression here. Are there diagnostics for ID2?

Reply:
ID2 is a project level no-location diagnostic that cannot be suppressed with editorconfig or pragma/SuppressMessAttribute. 
https://github.com/dotnet/roslyn/blob/bc641a6e58a5c15ac6b8052243c2b0dd97e65e1e/src/Compilers/Test/Core/Diagnostics/CommonDiagnosticAnalyzers.cs#L71-L72
and
https://github.com/dotnet/roslyn/blob/bc641a6e58a5c15ac6b8052243c2b0dd97e65e1e/src/Compilers/Test/Core/Diagnostics/CommonDiagnosticAnalyzers.cs#L41-L50

Below, the test invokes `GetExpectedV2ErrorLogRulesText` with a non-null `suppressionKinds1` and null `suppressionKinds2` to get the expected rules text with `ID1` suppressed and `ID2` unsuppressed. This helper method returns an expected entry for both ids, but suppression info added only for the ID with non-null `suppressionKinds` argument.